modules/solid_mechanics/doc/content/modules/solid_mechanics/1d_elastic_waves.md

```diff
@@ -1,5 +1,93 @@
+natural frequencies (eigenvalues) and mode shapes (eigenvectors) of a structure.
+The natural frequencies represent the frequencies at which the structure tends
+to vibrate when subjected to a disturbance, while the mode shapes describe the
+corresponding deformed shape of the structure at each natural frequency. Modal
+analysis provides valuable insights into the dynamic behavior of a structure and
+is often used for design, optimization, and troubleshooting purposes.
+
+The general eigenvalue problem that needs to be solved for modal analysis is given by:
+\begin{equation}
+(\mathbf{A} - \omega^2 \mathbf{B})\mathbf{u} = 0
+\end{equation}
+where $\mathbf{A}$ holds the "noneigen" kernels, $\mathbf{B}$ holds the "eigen" kernels, $\omega$ are the natural frequencies, and $\mathbf{u}$ are the mode shapes.
+In MOOSE, modal analysis is set up using the Eigenvalue Executioner. The
+Eigenvalue System allows you to define which kernels go into which matrix. It's important
+to note that in modal analysis, Neumann boundary conditions are
+ignored since we are interested in the free vibration response of the structure.
+Only Dirichlet boundary conditions are considered.
+
+To demonstrate modal analysis in MOOSE, we will visit the cantilever beam example that will be discussed in more detail in the next section on frequency response functions. The goal here is to calculate the first few natural frequencies and visualize the corresponding mode shapes of the cantilever beam.
+The strong form for the modal analysis of the cantilever beam is:
+\begin{equation}
+\nabla \cdot \boldsymbol{\sigma}(\mathbf{u}) = -\rho \omega^2 \mathbf{u}
+\end{equation}
+This equation represents the generalized eigenvalue problem for modal analysis.
+The terms on the left side, which involve the stress divergence, will contribute
+to the "A" matrix, and the terms on the right side, which are the mass terms, will contribute
+to the "B" matrix.
+
+## Modal Analysis: Cantilever Beam Example
+
+In the Kernels block, we define the kernels that contribute to the "A" and "B"
+matrices in the eigenvalue problem. The StressDivergenceTensor kernels represent the stress divergence terms and contribute to the "A"
+matrix. The ADCoefReaction kernels
+represent the mass terms and contribute to the "B" matrix. The
+[!param](/Kernels/CoefReaction/coefficient) in [kernels/CoefReaction.md] is set
+to a negative value which corresponds to a positive density. The
+`extra_vector_tags = 'eigen'` parameter is used to indicate that these kernels
+contribute to "B" matrix.
+
+!listing test/tests/modal_analysis/modal.i block=Kernels id=m_kernel caption=
+Kernels for A and B
+
+The BCs block defines the boundary conditions for the problem. In this example, we have Dirichlet boundary conditions applied to the left boundary for all displacement components. The [DirichletBC.md] is used for the "A" matrix, while the [EigenDirichletBC.md] is used for the "B" matrix.
```

Comment:
Would the boundary conditions on the B matrix ever be different from the A matrix?  If not, mention something about that.  It seems like there should have a eigen tag for BCs too instead of a separate BC.